server/controllers/user.js

```diff
@@ -34,45 +49,58 @@ exports.updatePersonalInformation = asyncHandler(async (req, res, next) => {
       { headline: headline, bio: bio, location: location },
       { new: true },
     );
+    profile = getProfile(user);
     res.status(202).json({
       success: true,
-      user: {
-        email: user.get('email'),
-        username: user.get('username'),
-        headline: user.get('headline'),
-        bio: user.get('bio'),
-        location: user.get('location'),
-        profilePicUrl: user.get('profilePicUrl'),
-      },
+      profile,
     });
   } catch (err) {
     res.status(400);
     throw new Error('Failed to update personal information');
   }
 });
 
-exports.updateProfilePicture = asyncHandler(async (req, res, next) => {
-  const { url } = req.body;
+exports.getUserInfo = asyncHandler(async (req, res, next) => {
+  const username = req.params.username;
+  try {
+    const user = await User.findOne({ username: username });
+    profile = getProfile(user).user;
+    res.status(200).json(profile);
+  } catch (err) {
+    res.status(400);
+    throw new Error('Failed to retrieve personal information');
+  }
+});
+exports.getContestsByUsername = asyncHandler(async (req, res, next) => {
+  const username = req.params.username;
```

Comment:
Same as above

Reply:
This is for getting the contests for the profile of any user, not just the logged in user.

Reply:
Makes perfect sense! Thanks for the clarification 😄